addons/purchase/tests/test_purchase.py

```diff
@@ -483,6 +483,46 @@ def test_purchase_multicurrency(self):
 
         self.assertEqual(order_b.order_line.price_unit, 10.0, 'The price unit should be 10.0')
 
+    def test_discount_update_when_no_seller_found(self):
+        """ Purchase order line should reset discount when no vendor found based on quantity
+        """
+        vendor = self.env['res.partner'].create({
+            'name': 'Test Vendor',
+            'supplier_rank': 1,
+        })
+
+        # Create product with one vendor and discount based on quantity
+        product = self.env['product.product'].create({
+            'name': 'Product',
+            'seller_ids': [
+                (0, 0, {
+                    'partner_id': vendor.id,
+                    'min_qty': 10,
+                    'price': 10,
+                    'discount': 10
+                }),
+                (0, 0, {
+                    'partner_id': vendor.id,
+                    'min_qty': 20,
+                    'price': 10,
+                    'discount': 15
+                })
+            ]
+        })
+
+        # Create new PO line with the product
+        po_form = Form(self.env['purchase.order'])
+        po_line = po_form.order_line.new()
+        po_line.product_id = product
```

Comment:
`Form` is a heavy mechanism, if we can avoid it, it's better
Something like:
```suggestion
    self.env['purchase.order'].with_company(company_a).create({
        'partner_id': self.partner_a.id,
        'order_line': [(0, 0, {
            'product_id': product.id,
            'product_qty': 1,
            'product_uom': product.uom_po_id.id,
            'price_unit': 1,
        })],
    })
```